war/src/main/java/executable/Main.java

```diff
@@ -99,56 +95,63 @@ public class Main {
-                    SUPPORTED_JAVA_VERSIONS, System.getProperty("java.specification.version"), System.getProperty("java.home"));
-            e.printStackTrace();
-        }
-    }
-
-    /*package*/ static void verifyJavaVersion(int javaClassVersion, boolean enableFutureJava) {
-        final String displayVersion = String.format("%d.0", javaClassVersion);
-        if (SUPPORTED_JAVA_CLASS_VERSIONS.contains(javaClassVersion)) {
+    /*package*/ static void verifyJavaVersion(int releaseVersion, boolean enableFutureJava) {
+        if (SUPPORTED_JAVA_VERSIONS.contains(releaseVersion)) {
             // Great!
-        } else if (javaClassVersion > MINIMUM_JAVA_CLASS_VERSION) {
+        } else if (releaseVersion >= MINIMUM_JAVA_VERSION) {
             if (enableFutureJava) {
-                LOGGER.log(Level.WARNING,
-                        String.format("Running with Java class version %s which is not in the list of supported versions: %s. " +
-                                        "Argument %s is set, so will continue. " +
-                                        "See https://jenkins.io/redirect/java-support/",
-                                javaClassVersion, SUPPORTED_JAVA_CLASS_VERSIONS, ENABLE_FUTURE_JAVA_CLI_SWITCH));
+                LOGGER.log(
+                        Level.WARNING,
+                        "Running with Java {0} from {1}, which is not yet fully supported."
+                            + " Continuing because {2} is set."
+                            + " Supported Java versions are: {3}."
+                            + " See https://jenkins.io/redirect/java-support/ for more information.",
+                        new Object[] {
+                            releaseVersion,
+                            System.getProperty("java.home"),
+                            ENABLE_FUTURE_JAVA_CLI_SWITCH,
+                            SUPPORTED_JAVA_VERSIONS,
+                        });
             } else {
-                Error error = new UnsupportedClassVersionError(displayVersion);
-                LOGGER.log(Level.SEVERE, String.format("Running with Java class version %s which is not in the list of supported versions: %s. " +
-                                "Run with the " + ENABLE_FUTURE_JAVA_CLI_SWITCH + " flag to enable such behavior. " +
-                                "See https://jenkins.io/redirect/java-support/",
-                        javaClassVersion, SUPPORTED_JAVA_CLASS_VERSIONS), error);
-                throw error;
+                throw new UnsupportedClassVersionError(
+                        String.format(
+                                "Running with Java %d from %s, which is not yet fully supported.%n"
+                                        + "Run the command again with the %s flag to enable preview support for future Java versions.%n"
+                                        + "Supported Java versions are: %s",
```

Comment:
This message might look a bit weird with Java 12…16 (phrased as if it's too new). If that's not a case we care about, that's fine.

Reply:
commit f296111fa7

```
$ java -jar war/target/jenkins.war
Running with Java 13 from /home/basil/Downloads/Direct/jdks/jdk-13.0.2+8, which is not fully supported.
Run the command again with the --enable-future-java flag to bypass this error.
Supported Java versions are: [11, 17]
See https://jenkins.io/redirect/java-support/ for more information.

$  java -jar war/target/jenkins.war
Running with Java 18 from /home/basil/Downloads/Direct/jdks/jdk-18.0.2+9, which is not yet fully supported.
Run the command again with the --enable-future-java flag to enable preview support for future Java versions.
Supported Java versions are: [11, 17]
See https://jenkins.io/redirect/java-support/ for more information.
```